opennms-config/src/main/java/org/opennms/netmgt/config/SnmpPeerFactory.java

```diff
@@ -77,83 +77,80 @@
  */
 public class SnmpPeerFactory implements SnmpAgentConfigFactory {
     private static final Logger LOG = LoggerFactory.getLogger(SnmpPeerFactory.class);
-    private static final ReadWriteUpdateLock m_globalLock = new ReentrantReadWriteUpdateLock();
-    private static final Lock m_readLock = m_globalLock.updateLock();
-    private static final Lock m_writeLock = m_globalLock.writeLock();
+
+    private static final int VERSION_UNSPECIFIED = -1;
+
+    private static File s_configFile;
 
     /**
      * The singleton instance of this factory
      */
-    private static SnmpPeerFactory m_singleton = null;
-
-    private static File m_configFile;
+    private static SnmpPeerFactory s_singleton = null;
 
     /**
-     * The config class loaded from the config file
+     * This member is set to true if the configuration file has been loaded.
      */
-    private SnmpConfig m_config;
+    private static volatile boolean s_loaded = false;
 
-    private static FileReloadContainer<SnmpConfig> m_container;
-    private static FileReloadCallback<SnmpConfig> m_callback;
+    private final ReadWriteUpdateLock m_globalLock = new ReentrantReadWriteUpdateLock();
```

Comment:
This change doesn't look correct. We're using the ReadWriteUpdateLock to protect access to the singleton so I think this lock needs to remain static. Can you explain any of the changes in this class?